docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,157 @@
+pool and another part is used as a reimbursement for the relay entry submitter.
+What percent of the fee goes to the group creation pool is a governable
+parameter.
+
+Given that we want to incentivize to submit a relay entry as soon as possible
+and that the price relation between gas price and T reimbursement for relay
+entry submission is unknown at the moment the fee is set by the governance, we
+should calculate a delay factor and delay penalty as in v1 of the random beacon.
+The delay penalty should be subtracted from reimbursement and sent to the
+sortition-pool-wide reward pool. The remaining part is sent to the relay entry
+submitter. Over the course 64x6 blocks, reimbursement goes to zero. On-chain we
+are not going to enforce any order of submitters but off-chain clients should
+agree to the order where the member with index 1 has 6 blocks to submit, after 6
+blocks member with index 2 tries to submit, after 12 blocks member with index 3
+tries to submit and so forth.
+
+Anyone should be able to submit a transaction creating a new group and this
+action should be detached from relay entry submission. The submitter of this
+transaction receives a part of the group creation pool and the remaining part
+should be reserved for DKG result submitter as reimbursement for future
+transaction. What percentage of the pool goes where is a governable parameter
+and should be determined based on the gas cost of the two transactions.
+
+To make this scheme works both relay entry and DKG result submission
+transactions need to have predictable gas cost. Although it is guaranteed for
+the latter, the former, in v1 of the beacon, depends on the gas cost of
+executing a callback and needs to be optimized.
+
+T rewards will be distributed continuously to all operators in the beacon
+sortition pool, just like in the case of tBTC v2.
+
+=== Callbacks
+
+In v1 of the random beacon, a callback is executed in the same transaction in
+which relay entry is submitted with a gas limit of 2M. This allows applications
+using the random beacon to avoid an additional complexity at their end but comes
+at the cost of random beacon operators who need to have enough ETH on their
+balance. This approach is not compatible with the idea that all rewards will be
+solely in T given that it is impossible to establish the cost of executing a
+callback in T.
+
+Instead of executing any callback in the same transaction in which relay entry
+is submitted, we can assume that only the simplest callbacks are allowed, up to
+some gas limit set by the governance. Applications wanting to use a relay entry
+should submit another transaction using the relay entry value previously set by
+the random beacon. They should also employ an additional security check making
+sure the entry submitted is only valid for a certain number of blocks to avoid
+situation when relay entry beacon transaction and application-specific
+transaction are executed far from each other.
```

Comment:
```suggestion
transactions are executed far from each other.
```

Reply:
Same explanation as for the previous comment.